compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/SemanticAnalyzer.java

```diff
@@ -521,6 +528,14 @@ public void visit(BLangVariableDef varDefNode) {
         if (varDefNode.var.expr == null && !types.defaultValueExists(varDefNode.pos, varDefNode.var.type)) {
             dlog.error(varDefNode.pos, DiagnosticCode.UNINITIALIZED_VARIABLE, varDefNode.var.name);
         }
+
+        if (varDefNode.var.type != null && varDefNode.var.type.tag == TypeTags.JSON) {
```

Comment:
lets move this to `visit(BLangConstrainedType constrainedTypeNode)` in SymbolEnter where the other validations are done.

Reply:
i think we don't need to check for `varDefNode.var.type != null`. If the type is not set for some vars, lets fix that..